### PR TITLE
support position_ids  as input for fa

### PR DESCRIPTION
**Pull request by shadow150519**

### CODE REVIEW

**Review by Seventeen17 — COMMENTED** [review]
It seems like we need to add patch for position_ids

**Comment by shadow150519** [comment]
> It seems like we need to add patch for position_ids

done. btw, in transformers version >= v4.47.0, `model_flash_attention_utils._flash_attention_forward` has changed its interface and add five params, but it seems old patch can still work. maybe need to change in the future.
new interface
```python
def _flash_attention_forward(
    query_states: torch.Tensor,
    key_states: torch.Tensor,
    value_states: torch.Tensor,
    attention_mask: torch.Tensor,
    query_length: int,
    is_causal: bool,
    dropout: float = 0.0,
    position_ids: Optional[torch.Tensor] = None,
    softmax_scale: Optional[float] = None,
    sliding_window: Optional[int] = None,
    use_top_left_mask: bool = False,
    softcap: Optional[float] = None,
    deterministic: bool = None,
    cu_seq_lens_q: Optional[torch.LongTensor] = None, # support passing fa params directly
    cu_seq_lens_k: Optional[torch.LongTensor] = None,
    max_length_q: Optional[int] = None,
    max_length_k: Optional[int] = None,
    target_dtype: Optional[torch.dtype] = None, # use for type conversion for peft
):
```

old interface
```python
def _flash_attention_forward(
    query_states: torch.Tensor,
    key_states: torch.Tensor,
    value_states: torch.Tensor,
    attention_mask: torch.Tensor,
    query_length: int,
    is_causal: bool,
    dropout: float = 0.0,
    position_ids: Optional[torch.Tensor] = None,
    softmax_scale: Optional[float] = None,
    sliding_window: Optional[int] = None,
    use_top_left_mask: bool = False,
    softcap: Optional[float] = None,
    deterministic: bool = None,
):
```

File: core-main-144598-1732534309
Comment (anw90) [inline core-main-144598-1732534309:1]: Remove the main and core dump files.

File: torchacc/utils/patch.py [hunk 1]
```diff
@@ -154,25 +153,39 @@ def _flash_attention_forward(
                 # TODO(to TianXing): support position_ids
```
Comment (anw90) [inline torchacc/utils/patch.py]: Remove the TODO

**Review by anw90 — APPROVED** [review]

**Pull request merged**